scipy/stats/_stats.pyx

```diff
@@ -300,3 +300,66 @@ def _weightedrankedtau(ordered[:] x, ordered[:] y, intp_t[:] rank, weigher, bool
     tau = ((tot - (v + u - t)) - 2. * exchanges_weight[0]
            ) / np.sqrt(tot - u) / np.sqrt(tot - v)
     return min(1., max(-1., tau))
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _gaussian_kernel_estimate(double[:, :] points, double[:, :] values,
+                              double[:, :] xi, double[:, :] precision):
+    """
+    def gaussian_kernel_estimate(double[:, :] points, double[:, :] values,
+                                  double[:, :] xi, double[:, :] precision)
+
+    Evaluate a multivariate Gaussian kernel estimate.
+
+    Parameters
+    ----------
+    points : double[:, :] with shape (n, d)
+        Data points to estimate from in d dimenions.
+    values : double[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : double[:, :] with shape (m, d)
+        Coordinates to evaluate the estimate at in d dimensions.
+    precision : double[:, :] with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        Multivariate Gaussian kernel estimate evaluated at the input coordinates.
+    """
+    cdef:
+        double[:, :] estimate
+        int i, j, k
+        int n = points.shape[0]
+        int d = points.shape[1]
+        int p = values.shape[1]
+        int m = xi.shape[0]
+        double arg, residual, norm
+
+    assert xi.shape[1] == d, "points and xi must have same trailing dim"
```

Comment:
You want to avoid using bare `assert`s because they will be removed if running Python with `-O`; see e.g.

https://docs.python.org/3/using/cmdline.html#cmdoption-o

Reply:
Good point, will change to raising `ValueError` explicitly.